hypothesis-python/src/hypothesis/searchstrategy/types.py

```diff
@@ -115,6 +116,11 @@ def from_typing_type(thing):
         elif len(elem_types) == 1 and elem_types[0] == ():
             return st.tuples()  # Empty tuple; see issue #1583
         return st.tuples(*map(st.from_type, elem_types))
+    if (
+        hasattr(typing, "Literal")
+        and getattr(thing, "__origin__", None) == typing.Literal
+    ):
+        return st.sampled_from(thing.__args__)
```

Comment:
Literals [can be nested](https://www.python.org/dev/peps/pep-0586/#legal-parameters-for-literal-at-type-check-time), and `Literal[Literal[Literal[1, 2, 3], "foo"], 5, None]` should be exactly equivalent to `Literal[1, 2, 3, "foo", 5, None]` - we'll need some extra logic to flatten this; I'd pull out an `is_typing_literal` function and then write the loop inline here.

We'll also need some tests that check nested literals are handled correctly.

Reply:
> We'll also need some tests that check nested literals are handled correctly.

Can I use a hypothesis strategy for this? Or would you recommend against it?

Reply:
You *could*, but I'd just use a traditional unit test.